src/mam4xx/convproc.hpp

```diff
@@ -1127,6 +1132,480 @@ void compute_wup(const int iconvtype, const int kk,
+  const int kk,     
+  const Real dt,   
+  const Real dt_u[/* nlev */],   
+  const Real dp_i[/* nlev */],   
+  const Real cldfrac_i[/* nlev */],      
+  const Real mu_p_eudp[/* nlev */],   
+  const Real aqfrac[ConvProc::pcnst_extd],         
+  const Real icwmr[/* nlev */],          
+  const Real rprd[/* nlev */],       
+  Real conu[/* nlev+1 */][ConvProc::pcnst_extd],           
+  Real dconudt_wetdep[/* nlev+1 */][ConvProc::pcnst_extd])
+{
+  // clang-format off
+  // -----------------------------------------------------------------------
+  //  compute tendency from wet deposition
+  // 
+  //     rprd               = precip formation as a grid-cell average (kgW/kgA/s)
+  //     icwmr              = cloud water MR within updraft area (kgW/kgA)
+  //     fupdr              = updraft fractional area (--)
+  //     A = rprd/fupdr     = precip formation rate within updraft area (kgW/kgA/s)
+  //     clw_preloss = cloud water MR before loss to precip
+  //                 = icwmr + dt*(rprd/fupdr)
+  //     B = A/clw_preloss  = (rprd/fupdr)/(icwmr + dt*rprd/fupdr)
+  //                        = rprd/(fupdr*icwmr + dt*rprd)
+  //                        = first-order removal rate (1/s)
+  //     C = dp/(mup/fupdr) = updraft air residence time in the layer (s)
+  // 
+  //     fraction removed = (1.0 - exp(-cdt)) where
+  //                  cdt = B*C = (fupdr*dp/mup)*[rprd/(fupdr*icwmr + dt*rprd)]
+  // 
+  //     Note1:  *** cdt is now sensitive to fupdr, which we do not really know,
+  //                 and is not the same as the convective cloud fraction
+  //     Note2:  dt is appropriate in the above cdt expression, not dtsub
+  // 
+  //     Apply wet removal at levels where
+  //        icwmr(k) > clw_cut  AND  rprd(k) > 0.0
+  //     as wet removal occurs in both liquid and ice clouds
+  // -----------------------------------------------------------------------
+  /*
+   cloudborne aerosol, so the arrays are dimensioned with pcnst_extd = pcnst*2
+   in :: doconvproc_extd[pcnst_extd] ! flag for doing convective transport
+   in :: kk                   ! vertical level index
+   in :: dt                   ! Model timestep [s]
+   in :: dt_u[pver]           ! lagrangian transport time in the updraft[s]
+   in :: dp_i[pver]            ! dp [mb]
+   in :: cldfrac_i[pver]      ! cldfrac at current (with adjustments) [fraction]
+   in :: mu_p_eudp[pver]      ! = mu_i[kp1] + eudp[k] [mb/s]
+   in :: aqfrac[pcnst_extd]   ! aqueous fraction of constituent in updraft [fraction]
+   in :: icwmr[pver]    ! Convective cloud water from zm scheme [kg/kg]
+   in :: rprd[pver]     ! Convective precipitation formation rate [kg/kg/s]
+   inout :: conu[pverp][pcnst_extd]   ! mix ratio in updraft at interfaces [kg/kg]
+   inout :: dconudt_wetdep[pverp][pcnst_extd] ! d(conu)/dt by wet removal[kg/kg/s]
+  */
+  // clang-format on
+  // cutoff value of cloud water for doing updraft [kg/kg]
+  const Real clw_cut = 1.0e-6;
```

Comment:
`BAD_CONSTANT`? Or at least something we should tag to revisit?

Reply:
Are we marking all non-attributed (no reference to a paper or anything else) constants as BAD_CONSTANT so we can revisit at a future date?

Reply:
I don't think we've been ultra-consistent about this. Certainly we mark physical constants this way. Error and "is there anything at all" thresholds less so (though given the lack of institutional knowledge, maybe we be). I'll leave this one to your judgement. The other threshold I saw in this PR is 1e-12, which to me is "really small in a numerical sense", so I'm fine not marking those.